Userland/Libraries/LibC/unistd.cpp

```diff
@@ -470,19 +475,30 @@ off_t lseek(int fd, off_t offset, int whence)
 }
 
 int link(const char* old_path, const char* new_path)
+{
+    return linkat(AT_FDCWD, old_path, AT_FDCWD, new_path, 0);
+}
+
+int linkat(int old_dirfd, const char* old_path, int new_dirfd, const char* new_path, int flags)
 {
     if (!old_path || !new_path) {
         errno = EFAULT;
         return -1;
     }
-    Syscall::SC_link_params params { { old_path, strlen(old_path) }, { new_path, strlen(new_path) } };
+    Syscall::SC_link_params params { old_dirfd, { old_path, strlen(old_path) }, new_dirfd, { new_path, strlen(new_path) }, flags };
     int rc = syscall(SC_link, &params);
     __RETURN_WITH_ERRNO(rc, rc, -1);
 }
 
 int unlink(const char* pathname)
 {
-    int rc = syscall(SC_unlink, pathname, strlen(pathname));
+    return unlinkat(AT_FDCWD, pathname, 0);
+}
+
+int unlinkat(int dirfd, const char* pathname, int flags)
+{
+    Syscall::SC_unlink_params params { dirfd, { pathname, strlen(pathname) }, flags };
+    int rc = syscall(SC_unlink, &params);
```

Comment:
Would it make sense to rename the syscall to `SC_unlinkat`, `sys$unlinkat`, etc.? Same for the other syscalls.

Reply:
I tried to follow the existing "precedent": `open(2)` has included the dirfd for a long time. I don't want to go into bikeshedding, but changing every filesystem-related syscall to have the `at` suffix would introduce pointless noise and confusion. But it's just my opinion :shrug: